google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java

```diff
@@ -1460,6 +1463,7 @@ public static Builder newBuilder() {
    * @return a [@code Blob} with complete information
    * @throws StorageException upon failure
    */
+  @Deprecated
```

Comment:
I would like to create a `RestartableInputStream` abstraction and create a new entry point that requires that. It would not have the drawbacks of `mark()`/`reset()`, although of course it would require users to understand a new concept. I think it's the least worst solution for this problem. I will do that later, though.